### PR TITLE
Standardize names

### LINKED ISSUE #28: Standardize names
As @lacava suggested in #22, couple TODO items:

- [ ] rename the datasets and feature names so that special characters, when possible, are replaced with `_`. Replace uppercase letters with lowercase letters.

- [ ] In addition, I think we should move all datasets one level up and remove the subdirectory `classification` and `regression` because this information is already in metadata to do before generating `metadata.yaml` files, unless this breaks too many things.

### LINKED ISSUE #31: move datasets to one folder?
commit 469bfb2932cd0755d9ca89dd80f6af81d6312d48 has the wine quality dataset in the datasets/ folder. as [discussed here](https://github.com/EpistasisLab/penn-ml-benchmarks/commit/0fbca9ca796517711ca89f34e57fc6228f10f9ef#commitcomment-39253437) because we have metadata defining tasks now, we could move all datasets to one folder instead of separate regression and classification folders.

### PR DESCRIPTION
**Pull request by weixuanfu**
For issue #28.

Closes #28, closes #31.

### CODE REVIEW

File: datasets/wine_quality_red/metadata.yaml [hunk 1]
```diff
@@ -17,22 +17,22 @@ target:
     Wine quality score based on sensory data, median of at least 3 evaluations made by wine experts.
   code: ranges from 0 (very bad) to 10 (very excellent)
 features: # list of features in the dataset
-  - name: fixed_acidity
+  - name: fixed acidity
```
Comment (trangdata) [inline datasets/wine_quality_red/metadata.yaml:20]: Is this an oversight or did we rename the feature names in this dataset? (I think the separator used to be a `.`)

**Review by trangdata — COMMENTED** [review]
Thank you, @weixuanfu for a thorough re-structure. I just have one question.

File: datasets/wine_quality_red/metadata.yaml (hunk 1 above)
Comment (weixuanfu) [reply datasets/wine_quality_red/metadata.yaml:20]: I changed the feature names based on [original dataset in pmlb1.0](https://github.com/EpistasisLab/penn-ml-benchmarks/blob/master/datasets/classification/wine-quality-red/README.md) . We may change them later if the dataset file ending in tsv.gz was updated based on [openml](https://www.openml.org/u/869). I think using "_" in feature name should be OK.

Comment (trangdata) [reply datasets/wine_quality_red/metadata.yaml:20]: Ah I see that you reverted the dataset back. Yup sounds good!

File: .github/workflows/pythonapp.yml [hunk 2]
```diff
@@ -25,7 +25,7 @@ jobs:
         python-version: 3.8
     - name: Install dependencies
       run: |
-        pip install nose numpy scipy scikit-learn pandas joblib pyyaml
+        pip install nose numpy scipy scikit-learn pandas joblib pyyaml parameterized
```
Comment (lacava) [inline .github/workflows/pythonapp.yml]: what is giving us this scikit-learn dependency? can it be removed?

**Review by lacava — APPROVED** [review]

File: .github/workflows/pythonapp.yml (hunk 2 above)
Comment (weixuanfu) [reply .github/workflows/pythonapp.yml]: Yep, I think it can be removed

Comment (weixuanfu) [reply .github/workflows/pythonapp.yml]: Also, joblib can be removed.

Comment (lacava) [reply .github/workflows/pythonapp.yml]: :+1:

**Pull request merged**